main.c

```diff
@@ -79,11 +80,11 @@ int main(int argc, char* argv[]) {
 
 /* handlers */
 
-#define invalid(k, v)                \
-	{                                \
-		templateSet(template, k, v); \
-		valid = false;               \
-	}
+static int invalid(Template* template, char* key, char* value) {
```

Comment:
The key and value should be immutable, right?

Reply:
Fixed. Pls chek again.